File: src/main/java/io/bloco/faker/components/Bool.java
package io.bloco.faker.components;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Bool extends FakerComponent {

    public Bool(FakerData data) {
        super(data);
    }

    public boolean bool() {
        return bool(0.5f);
    }

    public boolean bool(float trueRatio) {
        return Math.random() < trueRatio;
    }
}

File: lib/src/main/kotlin/io/bloco/faker/components/Bool.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData

class Bool(data: FakerData) : FakerComponent(data) {
    fun bool(trueRatio: Float = 0.5f): Boolean {
        return Math.random() < trueRatio
    }
}